tensorflow/contrib/opt/python/training/irprop_plus_test.py

```diff
@@ -0,0 +1,272 @@
+              grads0_sign = neg_grads0_np
+              grads1_sign = neg_grads1_np
+              # retract weight
+              update = neg_update
+              if context.executing_eagerly():
+                opt.apply_gradients(zip([-grads0, -grads1], [var0, var1]),
+                                    loss0_var)
+            else:
+              # pos grad
+              grads0_sign = grads0_np
+              grads1_sign = grads1_np
+              old_loss0 = loss0
+              update = pos_update
+
+              if t > 8:
+                loss0 = large_loss0_np
+                update = pos_large_update
+                if context.executing_eagerly():
+                  pos_large_update = opt.apply_gradients(zip([grads0, grads1],
+                                                             [var0, var1]),
+                                                         large_loss0_var)
+              else:
+                if context.executing_eagerly():
+                  opt.apply_gradients(zip([grads0, grads1], [var0, var1]),
+                                      loss0_var)
+
+            if not context.executing_eagerly():
+              self.evaluate(update)
+
+          var0_np, delta0, old_grads0, old_w_t0 = irprop_update_numpy(
+              var0_np,
+              old_w_t0,
+              grads0_sign,
+              old_grads0,
+              delta0,
+              eta_minus=eta_minus,
+              eta_plus=eta_plus,
+              delta_min=delta_min,
+              delta_max=delta_max,
+              error=loss0,
+              old_error=old_loss0)
+
+          var1_np, delta1, old_grads1, old_w_t1 = irprop_update_numpy(
+              var1_np,
+              old_w_t1,
+              grads1_sign,
+              old_grads1,
+              delta1,
+              eta_minus=eta_minus,
+              eta_plus=eta_plus,
+              delta_min=delta_min,
+              delta_max=delta_max,
+              error=loss0,
+              old_error=old_loss0)
+
+          self.assertAllCloseAccordingToType(var0_np, self.evaluate(var0))
+          self.assertAllCloseAccordingToType(var1_np, self.evaluate(var1))
+
+  def _testInvalidLoss(self):
+
+    for dtype in [dtypes.float32]:
+
+      loss_var_none = None
+      loss_var_vector = variables.Variable(
+          np.array([1.0], dtype=dtype.as_numpy_dtype))
+
+      var0_np = np.array([1.0, 2.0], dtype=dtype.as_numpy_dtype)
+      grads0_np = np.array([0.1, 0.1], dtype=dtype.as_numpy_dtype)
+
+      var0 = constant_op.constant(var0_np)
+      grads0 = constant_op.constant(grads0_np)
+
+      opt = irprop_plus.IRpropPlusOptimizer()
+      grads = zip([grads0], [var0])
+      self.assertRaises(ValueError, opt.apply_gradients, grads, loss_var_none)
+      self.assertRaises(ValueError, opt.apply_gradients, grads, loss_var_vector)
+      # type error in case no loss variable is passed, sanity check
+      self.assertRaises(TypeError, opt.apply_gradients, grads)
+
+  def testDense(self):
```

Comment:
Decorate this test with test_util.run_in_graph_and_eager_modes; as it is now it has a bunch of ifs related to eager which will never execute

Reply:
ok